improver_tests/wxcode/wxcode/test_ModalFromGroupings.py

```diff
@@ -0,0 +1,509 @@
+        # When snow shower codes are grouped, heavy snow shower is chosen as the
+        # snow shower codes are equally likely, so the first entry within the
+        # intensity category dictionary is chosen.
+        ([23, 23, 26, 26, 17, 17, 17, 17], True, 26, False),
+        # Demonstrate that reversing the ordering within the intensity categories
+        # gives a different result in the event of a tie.
+        ([23, 23, 26, 26, 17, 17, 17, 17], True, 23, True),
+        # Use ignore intensity option, with wet symbols that do not have intensity
+        # variants.
+        ([11, 11, 11, 11, 11, 11, 11, 11], True, 11, False),
+        # When snow shower codes are grouped, heavy snow shower is chosen as the
+        # snow shower codes are equally likely, so the first entry within the
+        # intensity category dictionary is chosen.
+        (
+            [[1, 1, 1, 1, 1, 1, 1, 1], [23, 23, 26, 26, 17, 17, 17, 17]],
+            True,
+            [1, 26],
+            False,
+        ),
+    ),
+)
+def test_expected_values_ignore_intensity(
+    wxcode_series, ignore_intensity, expected, reverse_intensity_dict
+):
+    """Test that the expected period representative symbol is returned."""
+    _, _, _, _, wxcode_cubes = wxcode_series
+    intensity_categories = INTENSITY_CATEGORIES.copy()
+    if reverse_intensity_dict:
+        intensity_categories = {}
+        for key in INTENSITY_CATEGORIES.keys():
+            intensity_categories[key] = [i for i in reversed(INTENSITY_CATEGORIES[key])]
+    result = ModalFromGroupings(
+        BROAD_CATEGORIES,
+        WET_CATEGORIES,
+        intensity_categories,
+        ignore_intensity=ignore_intensity,
+    )(wxcode_cubes)
+    expected = [expected] if not isinstance(expected, list) else expected
+    for index in range(len(expected)):
+        assert result.data.flatten()[index] == expected[index]
+
+
+@pytest.mark.parametrize("record_run_attr", [False])
+@pytest.mark.parametrize("model_id_attr", [False, True])
+@pytest.mark.parametrize("interval", [1])
+@pytest.mark.parametrize("offset_reference_times", [False, True])
+@pytest.mark.parametrize("cube_type", ["gridded", "spot"])
+@pytest.mark.parametrize(
+    "data, wet_bias, day_weighting, day_start, day_end, day_length, ignore_intensity, expected",
+    (
+        # The sleet code is the most common, so this is the modal code.
+        ([23, 23, 23, 26, 17, 17, 17, 17], 1, 1, 0, 12, 8, False, 17),
+        # The day weighting with the day start and day end set to same value has no
+        # impact on the modal code.
+        ([23, 23, 23, 26, 17, 17, 17, 17], 1, 10, 3, 3, 8, False, 17),
+        # The day weighting is set to emphasise the heavy snow shower (26).
+        ([23, 23, 23, 26, 17, 17, 17, 17], 1, 10, 4, 5, 8, False, 26),
+        # Without any weighting, there would be a dry symbol. A day weighting of 2
+        # results in 6 dry codes and 5 wet codes. A wet bias results in 6 dry codes
+        # and 10 wet codes.
```

Comment:
I don't think the wet bias changes the number of wet symbols anymore, it just changes the threshold required to choose a wet symbol as the summary symbol.

Reply:
I've modified this comment.